go/libraries/doltcore/doltdb/root_val.go

```diff
@@ -475,3 +477,53 @@ func (root *RootValue) RemoveTables(ctx context.Context, tables ...string) (*Roo
 
 	return newRootValue(root.vrw, rootValSt), nil
 }
+
+// DocDiff returns the slices of the dolt docs on the filesystem that are added, modified, and removed when compared with a root value.  Docs
+// In this instance that are not in the other instance are considered added, and docs in the other instance and not
+// this instance are considered removed.
+func (root *RootValue) DocDiff(ctx context.Context, newerLicenseBytes, newerReadmeBytes []byte) (added, modified, removed []string, err error) {
+	added = []string{}
+	modified = []string{}
+	removed = []string{}
+
+	docTable, found, err := root.GetTable(ctx, docTableName)
+
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	var rowMap types.Map
+	var licenseVal types.Value
+	var readmeVal types.Value
+	if found {
+		rowMap, err = docTable.GetRowData(ctx)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		licenseVal, _, err = rowMap.MaybeGet(ctx, types.String(licensePk))
+		readmeVal, _, err = rowMap.MaybeGet(ctx, types.String(readmePk))
+	}
+
+	if licenseVal == nil && newerLicenseBytes != nil {
```

Comment:
could convert this if / else if block into a function so that if you add a 3rd doc type you don't have to copy so much code.

Reply:
Moved this code to a helper function called `appendDocDiffs`: https://github.com/liquidata-inc/dolt/pull/222/files#diff-6f09c9d7a33cd9d8093aa6d760e03081R524